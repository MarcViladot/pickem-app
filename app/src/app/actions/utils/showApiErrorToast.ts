import {ResponseApiError, WebApiErrorCode} from '../../utils/IResponse';
import {ErrorTools} from '../../utils/ErrorTools';
import Toast from 'react-native-toast-message';
import i18next from 'i18next';

export const SHOW_TOAST = "SHOW_TOAST";
export const showApiErrorToast = (res: ResponseApiError) => {
    const translation = ErrorTools.getApiErrorMsgFromResponse(res);
    Toast.show({
        text1: i18next.t(translation.error, translation.paramsToTranslate),
        type: res.ErrorCode !== -2 ? 'apiError' : 'serverError',
        position: 'bottom',
        props: res
    });
    return {
        type: SHOW_TOAST
    }
};
