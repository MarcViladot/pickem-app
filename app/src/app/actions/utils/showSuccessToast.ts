import i18next from 'i18next';
import Toast from 'react-native-toast-message';

export const SHOW_SUCCESS_TOAST = 'SHOW_SUCCESS_TOAST';
export const showSuccessToast = (msg?: string) => {
  Toast.show({
    text1: msg || i18next.t('COMMON.OPERATION_OK'),
    type: 'success',
    position: 'bottom',
  });
  return {
    type: SHOW_SUCCESS_TOAST,
  };
};
