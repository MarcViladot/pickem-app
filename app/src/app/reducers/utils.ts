// import {SHOW_LOADING} from '../actions/utils/showLoading';
// import {HIDE_LOADING} from '../actions/utils/hideLoading';
// import {SET_LANG} from '../actions/utils/setLang';
import { AnyAction } from "redux";

const initialState = {
    serverLoading: false,
    currentLang: null
};

export default (state = initialState, action: AnyAction) => {
    /*if (action.type === SHOW_LOADING) {
        return {
            ...state,
            serverLoading: true
        };
    } else if (action.type === HIDE_LOADING) {
        return {
            ...state,
            serverLoading: false
        };
    } else if (action.type === SET_LANG) {
        return {
            ...state,
            currentLang: action.lang
        };
    }*/
    return state;
};
