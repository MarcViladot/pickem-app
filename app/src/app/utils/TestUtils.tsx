import store from '../../../store';
import {render} from '@testing-library/react-native'
import {Provider} from 'react-redux';
import React, {useState} from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {DarkTheme, LightTheme, ThemeContext} from '../../../themes';
import {ThemeProvider} from 'styled-components/native';

const AllTheProviders = ({children}) => {
    const [theme, setTheme] = useState('dark');
    const themeData = {theme, setTheme};
    return (
        <Provider store={store}>
            <ThemeContext.Provider value={themeData}>
                <ThemeProvider theme={theme === 'dark' ? DarkTheme : LightTheme}>
                    {children}
                </ThemeProvider>
            </ThemeContext.Provider>
        </Provider>
    )
};

const ReduxProvider = ({children}) => {

    return (
        <Provider store={store}>
            {children}
        </Provider>
    )
};

const ProvidersWithNav = ({children}) => {
    const [theme, setTheme] = useState('dark');
    const themeData = {theme, setTheme};
    return (
        <Provider store={store}>
            <ThemeContext.Provider value={themeData}>
                <ThemeProvider theme={DarkTheme}>
                    <NavigationContainer theme={DarkTheme}>
                        {children}
                    </NavigationContainer>
                </ThemeProvider>
            </ThemeContext.Provider>
        </Provider>
    )
};

const customRender = (ui, options?) =>
    render(ui, {wrapper: AllTheProviders, ...options});

const reduxRender = (ui, options?) =>
    render(ui, {wrapper: ReduxProvider, ...options});

const navRender = (ui, options?) =>
    render(ui, {wrapper: ProvidersWithNav, ...options});

export * from '@testing-library/react-native'
export {customRender as render}
export {reduxRender as reduxRender}
export {navRender as navRender}
