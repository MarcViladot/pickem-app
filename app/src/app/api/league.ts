import { User, UserCredentials } from "../interfaces/user.interface";
import Api from "./api";
import { AxiosError, AxiosResponse } from "axios";
import { ResponseApi, ResponseServerError } from "../utils/IResponse";
import {Round} from '../interfaces/league.interface';

export default {

  getGroupLeague(groupId: number, leagueId: number) {
    return Api.get(`league/getGroupLeague/${groupId}/${leagueId}`)
      .then((r: AxiosResponse<ResponseApi<any>>) => {
        return r.data
      }, (err: AxiosError) => {
        return new ResponseServerError(err);
      })
  },

  getRoundDetail(roundId: number) {
    return Api.get(`round/${roundId}`)
      .then((r: AxiosResponse<ResponseApi<Round>>) => {
        return r.data
      }, (err: AxiosError) => {
        return new ResponseServerError(err);
      })
  }
}
