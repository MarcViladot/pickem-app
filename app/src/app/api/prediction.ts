import { User, UserCredentials } from "../interfaces/user.interface";
import Api from "./api";
import { AxiosError, AxiosResponse } from "axios";
import { ResponseApi, ResponseServerError } from "../utils/IResponse";
import {CreateRoundPredictionDto, UpdatePredictionDto} from '../interfaces/round.interface';
import {Prediction, Round} from '../interfaces/league.interface';

export default {

    createRoundPrediction(createRoundPredictionDto: CreateRoundPredictionDto[]) {
        return Api.post(`prediction`, createRoundPredictionDto)
            .then((r: AxiosResponse<ResponseApi<Round>>) => {
                return r.data
            }, (err: AxiosError) => {
                return new ResponseServerError(err);
            })
    },

    updatePrediction(updatePredictionDto: UpdatePredictionDto) {
        return Api.put(`prediction`, updatePredictionDto)
            .then((r: AxiosResponse<ResponseApi<Prediction>>) => {
                return r.data
            }, (err: AxiosError) => {
                return new ResponseServerError(err);
            })
    }
}
