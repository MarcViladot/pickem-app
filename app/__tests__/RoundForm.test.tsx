import React from 'react';
import {render} from '../src/app/utils/TestUtils';
import {fireEvent, waitFor} from '@testing-library/react-native';
import RoundForm from '../src/app/components/common/RoundForm';
import {Round, TeamPosition} from '../src/app/interfaces/league.interface';
import prediction from '../src/app/api/prediction';

const mockRound: Round = {
    finished: false,
    id: 0,
    matches: [
        {
            id: 1,
            startDate: '2021-12-22T12:19:16.000Z',
            finished: false,
            doublePoints: false,
            teams: [
                {
                    id: 1,
                    finalResult: null,
                    teamPosition: TeamPosition.LOCAL,
                    team: {
                        id: 1,
                        name: 'Local team',
                        crest: '',
                    }
                },
                {
                    id: 2,
                    finalResult: null,
                    teamPosition: TeamPosition.AWAY,
                    team: {
                        id: 2,
                        name: 'Away team',
                        crest: '',
                    }
                }
            ],
            predictions: []
        }
    ],
    name: '',
    startingDate: '2021-12-22T12:19:16.000Z',
    translationGroup: {
        id: 1,
        groupName: 'round',
        roundNames: [
            {
                id: 1,
                text: 'Jornada',
                lang: 'es',
            },
            {
                id: 2,
                text: 'Round',
                lang: 'en',
            }
        ]
    },
    translationNameExtra: '1',
    visible: true
}

describe('Round form component', () => {

    it('renders correctly', async () => {
        const {toJSON} = await waitFor(() =>
            render(<RoundForm round={mockRound} onlyView={false} canEdit={false} canSubmit={true}
                              onSubmit={() => console.log('123')}
                              hasStarted={false}/>)
        );
        expect(toJSON()).toMatchSnapshot();
    });

    it('should render correctly the round pts if the round has started', async () => {
        const roundWithPredictions = JSON.parse(JSON.stringify(mockRound));
        roundWithPredictions.matches[0].predictions[0] = {
            id: 1,
            localTeamResult: 2,
            awayTeamResult: 2,
            correct: true,
            points: 11,
            userId: 1
        }
        const {findByText} = await waitFor(() =>
            render(<RoundForm round={roundWithPredictions} onlyView={false} canEdit={false} canSubmit={true}
                              onSubmit={() => console.log('123')}
                              hasStarted={true}/>)
        );
        expect(await findByText('11 pts')).toBeTruthy();
    });

    it('should render the submit button if the round has not started', async () => {
        const {findByText} = await waitFor(() =>
            render(<RoundForm round={mockRound} onlyView={false} canEdit={false} canSubmit={true}
                              onSubmit={() => console.log('123')}
                              hasStarted={false}/>)
        );
        expect(await findByText('Submit')).toBeTruthy();
    });

    it('should render the match points if match has finished', async () => {
        const roundWithPredictions = JSON.parse(JSON.stringify(mockRound));
        roundWithPredictions.matches[0].predictions = [{
            id: 1,
            localTeamResult: 2,
            awayTeamResult: 2,
            correct: true,
            points: 11,
            userId: 1
        }]
        roundWithPredictions.matches[0].finished = true;
        const {findByText} = await waitFor(() =>
            render(<RoundForm round={roundWithPredictions} onlyView={false} canEdit={false} canSubmit={true}
                              onSubmit={() => console.log('123')}
                              hasStarted={true}/>)
        );
        expect(await findByText('11 Pts')).toBeTruthy();
    });

    it('should increment prediction on increment button click', async () => {
        const {findByTestId} = await waitFor(() =>
            render(<RoundForm round={mockRound} onlyView={false} canEdit={false} canSubmit={true}
                              onSubmit={() => console.log('123')}
                              hasStarted={false}/>)
        );
        const incrementButton = await findByTestId('IncrementAwayButton');
        const field = await findByTestId('AwayField');
        fireEvent.press(incrementButton);
        expect(field.props.value).toBe('0');
    });

    it('should increment prediction on increment button click', async () => {
        const roundWithPredictions = JSON.parse(JSON.stringify(mockRound));
        roundWithPredictions.matches[0].predictions = [{
            id: 1,
            localTeamResult: 0,
            awayTeamResult: 5,
            correct: true,
            points: 11,
            userId: 1
        }]
        const {findByTestId} = await waitFor(() =>
            render(<RoundForm round={roundWithPredictions} onlyView={false} canEdit={false} canSubmit={true}
                              onSubmit={() => console.log('123')}
                              hasStarted={false}/>)
        );
        const incrementButton = await findByTestId('DecrementAwayButton');
        const field = await findByTestId('AwayField');
        expect(field.props.value).toBe('5');
        fireEvent.press(incrementButton);
        expect(field.props.value).toBe('4');
    });

    it('should render "-" if prediction is not set', async () => {
        const {findByTestId} = await waitFor(() =>
            render(<RoundForm round={mockRound} onlyView={false} canEdit={false} canSubmit={true}
                              onSubmit={() => console.log('123')}
                              hasStarted={false}/>)
        );
        const field = await findByTestId('AwayField');
        expect(field.props.value).toBe('-');
    });

    it('should render prediction if it is set', async () => {
        const roundWithPredictions = JSON.parse(JSON.stringify(mockRound));
        roundWithPredictions.matches[0].predictions = [{
            id: 1,
            localTeamResult: 1,
            awayTeamResult: 123,
            correct: true,
            points: 11,
            userId: 1
        }]
        const {findByTestId} = await waitFor(() =>
            render(<RoundForm round={roundWithPredictions} onlyView={false} canEdit={false} canSubmit={true}
                              onSubmit={() => console.log('123')}
                              hasStarted={false}/>)
        );
        const field = await findByTestId('AwayField');
        expect(field.props.value).toBe('123');
    });

    it('should not call createRoundPredictions if predictions have not been set', async () => {
        const {findByTestId} = await waitFor(() =>
            render(<RoundForm round={mockRound} onlyView={false} canEdit={false} canSubmit={true}
                              onSubmit={() => console.log('123')}
                              hasStarted={false}/>)
        );
        const spyOn = jest.spyOn(prediction, 'createRoundPrediction');
        const submitButton = await findByTestId('SubmitButton');
        fireEvent.press(submitButton);
        expect(spyOn).not.toHaveBeenCalled();
    });

    it('should call createRoundPredictions if predictions have been set', async () => {
        const {findByTestId} = await waitFor(() =>
            render(<RoundForm round={mockRound} onlyView={false} canEdit={false} canSubmit={true}
                              onSubmit={() => console.log('123')}
                              hasStarted={false}/>)
        );
        const spyOn = jest.spyOn(prediction, 'createRoundPrediction');
        const incrementAwayButton = await findByTestId('IncrementAwayButton');
        const incrementLocalButton = await findByTestId('IncrementLocalButton');
        fireEvent.press(incrementAwayButton);
        fireEvent.press(incrementLocalButton);
        const submitButton = await findByTestId('SubmitButton');
        fireEvent.press(submitButton);
        await waitFor(() => expect(spyOn).toHaveBeenCalled());
    });

    it('should render edit button if round is editable and prediction is set', async () => {
        const roundWithPredictions = JSON.parse(JSON.stringify(mockRound));
        roundWithPredictions.matches[0].predictions = [{
            id: 1,
            localTeamResult: 1,
            awayTeamResult: 123,
            correct: true,
            points: 11,
            userId: 1
        }]
        const {findByTestId} = await waitFor(() =>
            render(<RoundForm round={roundWithPredictions} onlyView={false} canEdit={true} canSubmit={false}
                              onSubmit={() => console.log('123')}
                              hasStarted={false}/>)
        );
        const button = await findByTestId('EditButton');
        expect(button).toBeTruthy();
    });

    it('should not render buttons if round is only viewable even if user can edit or submit', async () => {
        const {queryByTestId} = await waitFor(() =>
            render(<RoundForm round={mockRound} onlyView={true} canEdit={true} canSubmit={true}
                              onSubmit={() => console.log('123')}
                              hasStarted={false}/>)
        );
        const submitButton = await queryByTestId('SubmitButton');
        const incrementAwayButton = await queryByTestId('IncrementAwayButton');
        const incrementLocalButton = await queryByTestId('IncrementLocalButton');
        expect(submitButton).toBeFalsy();
        expect(incrementAwayButton).toBeFalsy();
        expect(incrementLocalButton).toBeFalsy();
    });

});
