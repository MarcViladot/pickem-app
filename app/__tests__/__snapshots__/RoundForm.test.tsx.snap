// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Round form component renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 40,
      "padding": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#1F2C39",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
          "padding": 7,
        },
      ]
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#FFF",
            },
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 5,
            },
          ]
        }
      >
        Round 1
      </Text>
      <Text
        style={
          Object {
            "color": "#A6A6A6",
            "fontSize": 13,
          }
        }
      >
        STARTING AT 
        22/12 13:19
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#CCCCCC",
          "borderRadius": 50,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 100,
        }
      }
      testID="SubmitButton"
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontSize": 17,
          }
        }
      >
        Submit
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#1F2C39",
        },
        Object {
          "padding": 7,
          "paddingBottom": -5,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "marginBottom": 12,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 2,
              "padding": 7,
            },
            Object {
              "backgroundColor": "#F4F4F4",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFF",
                },
                Object {
                  "fontSize": 12,
                },
              ]
            }
          >
            AT
             
            22/12 13:19
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
              "marginBottom": 1,
              "paddingHorizontal": 15,
            },
            Object {
              "backgroundColor": "#F4F4F4",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "",
              }
            }
            style={
              Object {
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#FFF",
                },
              ]
            }
          >
            Local team
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 0.5,
              }
            }
            testID="DecrementLocalButton"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={12}
              bbWidth={12}
              fill="#464646"
              focusable={false}
              height={12}
              meetOrSlice={0}
              minX={0}
              minY={0}
              secondaryFill={null}
              secondaryOpacity={0.4}
              size={12}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {},
                  Object {
                    "flex": 0,
                    "height": 12,
                    "width": 12,
                  },
                ]
              }
              vbHeight={512}
              vbWidth={448}
              width={12}
            >
              <RNSVGGroup
                fill={4282795590}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                  fill={4282795590}
                  fillOpacity={1}
                  propList={
                    Array [
                      "fill",
                      "fillOpacity",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <TextInput
            editable={false}
            editing={true}
            style={
              Array [
                Object {
                  "backgroundColor": "#FFF",
                  "borderColor": "#A6A6A6",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "color": "black",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "height": 40,
                  "marginBottom": 0,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 0,
                  "textAlign": "center",
                  "width": 30,
                },
              ]
            }
            testID="LocalField"
            value="-"
          />
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="IncrementLocalButton"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={12}
              bbWidth={12}
              fill="#464646"
              focusable={false}
              height={12}
              meetOrSlice={0}
              minX={0}
              minY={0}
              secondaryFill={null}
              secondaryOpacity={0.4}
              size={12}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {},
                  Object {
                    "flex": 0,
                    "height": 12,
                    "width": 12,
                  },
                ]
              }
              vbHeight={512}
              vbWidth={448}
              width={12}
            >
              <RNSVGGroup
                fill={4282795590}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                  fill={4282795590}
                  fillOpacity={1}
                  propList={
                    Array [
                      "fill",
                      "fillOpacity",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
              "marginBottom": 1,
              "paddingHorizontal": 15,
            },
            Object {
              "backgroundColor": "#F4F4F4",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "",
              }
            }
            style={
              Object {
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#FFF",
                },
              ]
            }
          >
            Away team
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 0.5,
              }
            }
            testID="DecrementAwayButton"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={12}
              bbWidth={12}
              fill="#464646"
              focusable={false}
              height={12}
              meetOrSlice={0}
              minX={0}
              minY={0}
              secondaryFill={null}
              secondaryOpacity={0.4}
              size={12}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {},
                  Object {
                    "flex": 0,
                    "height": 12,
                    "width": 12,
                  },
                ]
              }
              vbHeight={512}
              vbWidth={448}
              width={12}
            >
              <RNSVGGroup
                fill={4282795590}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                  fill={4282795590}
                  fillOpacity={1}
                  propList={
                    Array [
                      "fill",
                      "fillOpacity",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <TextInput
            editable={false}
            editing={true}
            style={
              Array [
                Object {
                  "backgroundColor": "#FFF",
                  "borderColor": "#A6A6A6",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "color": "black",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "height": 40,
                  "marginBottom": 0,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 0,
                  "textAlign": "center",
                  "width": 30,
                },
              ]
            }
            testID="AwayField"
            value="-"
          />
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="IncrementAwayButton"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={12}
              bbWidth={12}
              fill="#464646"
              focusable={false}
              height={12}
              meetOrSlice={0}
              minX={0}
              minY={0}
              secondaryFill={null}
              secondaryOpacity={0.4}
              size={12}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {},
                  Object {
                    "flex": 0,
                    "height": 12,
                    "width": 12,
                  },
                ]
              }
              vbHeight={512}
              vbWidth={448}
              width={12}
            >
              <RNSVGGroup
                fill={4282795590}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                  fill={4282795590}
                  fillOpacity={1}
                  propList={
                    Array [
                      "fill",
                      "fillOpacity",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
